distribution/std-lib/Standard/src/Image/Data/Matrix.enso

```diff
@@ -1,11 +1,23 @@
 from Standard.Base import all
+
 import Standard.Image.Data.Histogram
 import Standard.Image.Data.Image
 import Standard.Image.Data.Matrix.Internal
 
 polyglot java import org.enso.image.data.Matrix as Java_Matrix
 
-type Index_Out_Of_Bounds_Error index
+type Matrix_Error
+
+    ## Indicates that a matrix has been accessed with an illegal index.
+    type Index_Out_Of_Bounds_Error index
+
+    ## Indicates that operation has failed.
+    type Dimensions_Not_Equal
+
+    to_display_text : Text
+    to_display_text = case this of
+        Index_Out_Of_Bounds_Error index -> 'The index ' + index ' is out of bounds.'
```

Comment:
This should say "for a matrix with dimensions ...". Also please make sure you test all of these. This won't work as you need to call `.to_text` on `index`.